### DISCUSSION TITLE
Malilib.jar by sakura-ryoko has a trojan

### DISCUSSION THREAD

**Discussion by MykalJacsun** [maruohon/malilib#156] [General]
Sakura ryokos malilib 1.21 sakura2 or whatever its called has a PUA virus on it wow amazing. thankfully as it is a jar the virus is not too hard to remove just perma delete the jar however it most likely copied itself to appdata so make sure to look around in file explorer for anything suspicious. would not recommend downloading any of sakura-ryokos developments.

**Comment by MykalJacsun** [comment]
the litematica jar also has a trojan

**Comment by MykalJacsun** [comment]
![Screenshot 2024-06-17 173307](https://github.com/maruohon/malilib/assets/121336831/1b55d57c-0dfc-4251-b21e-45c0165f888e)
![Screenshot 2024-06-17 173441](https://github.com/maruohon/malilib/assets/121336831/9b51c46e-50d0-4fb6-a652-e6cd8bff92dd)

**Reply by mlaikofta222** [reply]
I don't think the UUID is that private - but if you're going to censor something private, make sure it's super censored.

"add???5e-5c?5-4be6-b527-???????f72"
Somewhat censored, but stay safe :)

**Comment by MykalJacsun** [comment]
the malilib jar doesnt have the trojan sorry for the incorrect title but basically the same effect

**Comment by sakura-ryoko** [comment]
It's a false positive affecting many mods, if you got it from my Github, its safe to mark it as "safe" .  It does not have a "trojan" .

**Comment by MykalJacsun** [comment]
Only mods from their page triggered it out of tens of 1.21 mods I
downloaded? Seems unlikely

**Comment by ThirtyTwelveNor** [comment]
sounds like a you problem. 
[VirusTotal malilib-fabric-1.21-0.19.999-sakura.2](https://www.virustotal.com/gui/file/66b2984079d9a783e5170d37186a6dbb0e438e81be1149092996f30be5fd111e)
[VirusTotal litematica-fabric-1.21-0.18.999-sakura.3.jar](https://www.virustotal.com/gui/file/d0419ad53f7cb0fc93d1b47f870d5976636edae2f66f7ebd2d3be40fc952dad9)
[VirusTotal tweakeroo-fabric-1.21-0.20.999-sakura.2.jar](https://www.virustotal.com/gui/file/1b69214598bc49721863ec60d9a718028da07a6062c500d5dddcfd9fba8da89a)
[VirusTotal minihud-fabric-1.21-0.31.999-sakura.3.jar](https://www.virustotal.com/gui/file/b4227038c0b2239f9f1200a64e0923ec0a45704597977bff2b2ff41ba523cda7)

**Comment by TymonMarek** [comment]
Many Windows Defender alerts that end with `!ml` (Machine Learning Detection) are usually false positives. These are detections from Microsoft's AI.

![image](https://github.com/maruohon/malilib/assets/117597583/da0a518f-6e9d-40f1-97c4-21d250664750)

**Comment by benhovinga** [comment]
All of the latest versions of @sakura-ryoko fork are flagged with a trojan on Virus Total.

I downloaded and tested the last 5 releases from their github and all contain the same virus flag `Trojan.PY.Agent.AG`

- [VirusTotal malilib-fabric-1.21.5-0.24.0-sakura.8.jar](https://www.virustotal.com/gui/file/ac2975b58a8139f8affd0cba3057417b9f0bf703ab95284b94485a06d663cd9d)
- [VirusTotal malilib-fabric-1.21.5-0.24.0-sakura.7.jar](https://www.virustotal.com/gui/file/bf69e011a6f797898008e6467bbeb4e88d5594c5a370268c9d23bd7db583bde9)
- [VirusTotal malilib-fabric-1.21.4-0.23.3-sakura.3.jar](https://www.virustotal.com/gui/file/41fd458839d681091a8e25efdb60a36c505e5cc840ac6db6467d9e7f2d71e198)
- [VirusTotal malilib-fabric-1.21.5-0.24.0-sakura.6.jar](https://www.virustotal.com/gui/file/86fbbe77059d4fee5753b403dff201bb479244375930d55052e65afa80c59cc1)
- [VirusTotal malilib-fabric-1.21.5-0.24.0-sakura.5.jar](https://www.virustotal.com/gui/file/45abd858852f020e6ba617466d479c5295824596d1bae66d9ec499e4bf4b13e6)

**Comment by sakura-ryoko** [comment]
False positives yet again.

**Comment by mlaikofta222** [comment]
> "False positives yet again."



So why do many different antivirus platforms report a trojan? Only the structure of 1, maybe a couple, antiviruses might flag the same false positives. For all the innocent players out there - I'm not saying that the Trojan is real, but stay on the safer side - get the official release by maruohon. The official creator is even conveniently listed in the URL to this forum page! There's no reason not to.

Edit: Also, I am aware that maruohon stopped the updates, but would you rather have litematica _and_ a POSSIBLE (not saying it's real, just that it's likely) Trojan horse, or neither?

**Reply by sakura-ryoko** [reply]
> > "False positives yet again."
> 
>  - get the official release by maruohon. The official creator is even conveniently listed in the URL to this forum page! There's no reason not to.

I agree.  The `-sakura.X` builds are only meant for beta testing, etc; and people should be getting them from the Official Modrinth pages after I upload them; or whenever Masa uploads them to his Curse Forge page on occasion.
Technically speaking; the `-sakura.X` builds are also "Official" from my Github only.  If they appear anywhere else; they should be checked by Hash Code or downloaded from my Github.

I often tell people to go and check the code if they don't agree or build them yourself from my Github if you don't trust my builds for whatever reason; or find the "Virus" in the source code.

**Reply by mlaikofta222** [reply]
Not accusing you - just trying to stay safe. Sorry, please don't be offended :)

**Reply by sakura-ryoko** [reply]
> Not accusing you - just trying to stay safe. Sorry, please don't be offended :)

Part of what's happening is people's web browsers are using AI to "scan" files and often flag items that it doesn't recognize; or if people "re-post" my code with something that shouldn't be there.  If the files are not named with the Official naming conventions; that will be a clear sign that someone modified it; or checking them by Hash Code.

**Reply by mlaikofta222** [reply]
Also, 0.27.3 isn't flagged - just the newest ones.

**Reply by sakura-ryoko** [reply]
> Also, 0.27.3 isn't flagged - just the newest ones.

There seems to be no rhyme or reason to it.  I can build them exactly the same way and one might get flagged; and the other won't.  It doesn't matter if I use the build script or build them on my computer; or my work computer; the same can happen.

**Comment by TymonMarek** [comment]
If you would like to test this yourself, you can diff the release before Windows Defender marked it as a trojan and the release in which it started happening. This will show you what was added in that release and therefore what Windows Defender may be flagging as "malicious behaviour".

I feel like this issue should be closed since it is not a real issue per-se.
